stl_text/ops/vocab/pytext_vocab.py

```diff
@@ -34,42 +81,42 @@ def __init__(
         ), "vocab_path and vocab_list are mutual exclusive"
 
         if vocab_list:
-            self.vocab = ScriptVocabulary(vocab_list)
+            self.vocab = build_vocab_from_iterator([vocab_list],unk_token=UNK)
         else:
-            with PathManager.open(vocab_path) as f:
-                vocab = build_fairseq_vocab(
-                    f, special_token_replacements=special_token_replacements
-                )
-                self.vocab = ScriptVocabulary(
-                    list(vocab),
-                    pad_idx=vocab.get_pad_index(-1),
-                    bos_idx=vocab.get_bos_index(-1),
-                    eos_idx=vocab.get_eos_index(-1),
-                    unk_idx=vocab.get_unk_index(-1),
-                    unk_token=vocab.unk_token,
-                )
+            path_manager = PathManager()
+            with path_manager.open(vocab_path, "r",encoding='utf-8') as f:
+                self.vocab = load_vocab_from_file(f,unk_token=UNK)
+ 
+        self.vocab.append_token(BOS)
+        self.vocab.append_token(EOS)
+        self.vocab.append_token(PAD)
+        self.vocab.append_token(MASK)
+
+        self.bos_idx = self.vocab([BOS])[0]
+        self.eos_idx = self.vocab([EOS])[0]
+
         # TODO T77728853 We need to combine truncate with BOS/EOS as they impact each other
         # Need to find a nicer way to do this, as this can't be chained.
         self.add_bos = add_bos
         self.add_eos = add_eos
         # Make room for bos and eos from max_seq_len if true
         self.truncate_transform = TruncateTransform(max_seq_len - add_bos - add_eos)
 
+        self.vocab = self.vocab.to_ivalue() #Remove to_ivalue() PR: https://github.com/pytorch/text/pull/1080
```

Comment:
On the eager mode, we suggest to use the pybind vocab (aka, not to call `to_ivalue`) for better performance. `to_ivalue` is called to switch to torchbind vocab before calling JIT.

Reply:
> On the eager mode, we suggest to use the pybind vocab (aka, not to call `to_ivalue`) for better performance. `to_ivalue` is called to switch to torchbind vocab before calling JIT.

Right, i also notice similar behavior when doing the benchmarking...

Reply:
Same as my comment above.